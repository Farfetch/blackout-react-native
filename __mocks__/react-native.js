export const NativeModules = {
  SettingsManager: {
    settings: {
      AppleLocale: 'en',
      AppleLanguages: ['pt-PT', 'en'],
    },
  },
  I18nManager: {
    localeIdentifier: 'en',
  },
};

export const Platform = {
  OS: 'ios',
};

export const Dimensions = {
  get: dim => {
    if (dim === 'window') {
      return {
        width: 800,
        height: 1000,
      };
    }

    if (dim === 'screen') {
      return {
        width: 800,
        height: 1100,
      };
    }
  },
};
