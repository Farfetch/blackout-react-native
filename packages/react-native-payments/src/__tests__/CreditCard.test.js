/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react-native';
import { CreditCard } from '../';

jest.mock('react-native-webview');

jest.mock('@farfetch/blackout-react/authentication/hooks', () => ({
  useAuthentication: () => ({
    activeTokenData: {
      data: { accessToken: 'mock_guid' },
    },
  }),
}));

describe('CreditCard', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const mockDispatch = jest.fn();

  const props = {
    additionalProperies: {
      url: 'mockurl',
      staticName: 'whitelabel',
      folderName: 'pg-12',
      locale: 'en-PT',
    },
    creditCardDispatch: mockDispatch,
    paymentIntentId: '48f9cd72-7fcb-42e6-b610-b870ec5343ec',
    webViewRef: () => ({ current: { injectJavaScript: jest.fn() } }),
  };

  it('should render the WebView with a correct source url', () => {
    const { toJSON } = render(<CreditCard {...props} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should call dispatch', () => {
    render(<CreditCard {...props} />);

    expect(mockDispatch).toHaveBeenCalledWith({
      instrumentAdded: true,
      instrumentId: '48f9cd72-7fcb-42e6-b610-b870ec5343ec',
    });
  });
});
