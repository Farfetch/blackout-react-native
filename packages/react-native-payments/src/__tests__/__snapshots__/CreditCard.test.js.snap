// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditCard should render the WebView with a correct source url 1`] = `
<div>
  mockurl?paymentIntentId=48f9cd72-7fcb-42e6-b610-b870ec5343ec&staticName=whitelabel&folderName=pg-12&locale=en-PT
</div>
`;
